### DISCUSSION TITLE
Option for cookie consent to be excluded from block list

### DISCUSSION THREAD

**Discussion by metablaster** [uBlockOrigin/uAssets#15606] [Ideas]
### Prerequisites

- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues) to avoid opening a duplicate issue
- [X] I am running the latest version of uBlock Origin

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### URL(s) where the issue occurs

`turnbasedlovers.com`
`www.tenforums.com`

### Describe the issue

Many websites provide a cookie consent dialog where a user can choose which cookies to allow and adjust privacy options per domain.

uBlock static filters work in such a way where those consent dialogs are blocked, on some sites I'm presented with cookie consent but on most sites consent is blocked.

The issue is that those cookie consent dialogs aren't always something which one doesn't want to see, I'm one of those who want to opt in and adjust cookie settings per domain, however that's impossible without disabling almost all static filters which is not an option since that's same as disabling uBlock origin all togethter.

Thus I suggest that uBlock provides an option which would let ignore static filters when it come to blocking cookie consent on domains.
I suppose this has nothing to do with static filters but rather about how uBlock origin uses and applies filters, it certainly doesn't handle cookie consent separately.

### Screenshot(s)

Cookie consent dialog example:

 ![Image](https://i.imgur.com/uEbOhAI.png)

My settings:
![Image](https://i.imgur.com/a5pZbSd.png)

### uBlock Origin version

1.45.0

### Browser name and version

MS-Edge 107.0.1418.35 (Official build) (64-bit)

### Settings

Enabled all AdBlock static filters
Enabled all privacy and malware static filters
Enabled "Peter Lowe’s Ad and tracking server list" static filter

Other settings are displayed in screenshot.

None of the advanced user settings have been modified.

### Notes

`turnbasedlovers.com` site above is one example which you can test.
Default uBlock static filter which will reproduce the problem is called "uBlock filters" under "Built-in" group, if enabled cookie consent will be blocked.

**Comment by MasterKia** [comment]
The default uBO filter lists **do not** remove any cookie consent notices.

> Enabled all AdBlock static filters

You should reset uBO to default settings and not enable filter lists such as `EasyList Cookie` or `Fanboy/AdGuard Annoyances` afterwards.

**Comment by metablaster** [comment]
I have just reset uBlock in firefox to default and tested as you said, this works for `turnbasedlovers.com` site.
However it doesn't work for `www.tenforums.com`

The culprit is default `uBlock filters` under `Built-in` group as I described.
can you test it to confirm?

**Comment by Yuki2718** [comment]
That is because the site relies on `fundingchoicesmessages.google.com` which is also used to display anti-adblock to show cookie consent. Unblocking this means user will see so-called Google Funding Choice anti-adblock everywhere where the admin set [ad blocking recovery message](https://support.google.com/adsense/answer/11576085?hl=en), obviously not an option. Some other sites rely on tracker to show cookie consent. Anyway this is not an issue to be addressed in Issue ticket, possibly for [Discussions](https://github.com/uBlockOrigin/uAssets/discussions).

**Comment by mapx-** [comment]
@metablaster 

you could add in your "my filters"

```
@@||fundingchoicesmessages.google.com$script,domain=tenforums.com
tenforums.com##.adsbygoogle:upward(1)
```

**Comment by metablaster** [comment]
Thank you making this a discussion because I have one question.
Let's suppose some static filter blocks a cookie consent such as in this case due to google funding and similar...

Then here is an example of a cookie consent (which get's blocked) which sets some default choices like in the screenshot below:
from `turnbasedlovers.com` example in OP.

![Image](https://i.imgur.com/tA0HaVQ.png)

As you can see some default options are set upfront.
What I don't understand here, if this consent is blocked, isn't this the same thing as me accepting these default choices?
Obviously I could have uncheck all those choices if the consent remained, so what exactly happens with those choices, are they auto-confirmed and a cookie is set, or what exactly happens?


@mapx-
thanks for suggestion but I didn't know this is due to google thing, so I'm not going to allow it.

**Reply by Yuki2718** [reply]
Okay tested on Chrome, in `www.tenforums.com` case no additional cookie will be set with default uBO settings - three cookies (bb_forum_view, bb_lastactivity, bb_lastvisit) are always set regardless of consent. If you allow fundingchoices and accept all cookies, two additional cookies (FCNEC and FCCDCF) will be set. If you choose "Manage options" in the dialogue and either choose "Confirm choices" or manually decline all "Legitimate interest" (this was veerry long list under "Vendor preferences". I curse Google for not providing "Uncheck all" button.), only FCCDCF cookie will additonally be set. I don't know about `turnbasedlovers.com` case, you can check with browser's dev tools.

**Comment by metablaster** [comment]
Thanks for testing, so basically if consent is blocked, from web server perspective that's the same thing as user not interacting with the consent at all.

I guess it then depends on how the web server will respond to user not interacting with the consent, that is, there are only 2 options:
1. it may set cookies according to default options of the consent
2. or not set specific cookies at all (beyond those which are always set regardless of consent)

In other words, default options if present also handle a case when a user doesn't interact with the consent?
Can you agree with this hypothesis?

Surely I could test various web sites but there is no quarantee that every domain will handle the case in same way, and also whether a cookie is set or not doesn't tell much, data withing cookie is what matters, but we can't understand what does this data imply without further testing, so that's a bit too much to test.
